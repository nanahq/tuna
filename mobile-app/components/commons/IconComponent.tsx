import React from 'react'

import {IconProps } from '@expo/vector-icons/build/createIconSet'
import {
    MaterialIcons,
    Feather,
    MaterialCommunityIcons,
    AntDesign,
    Foundation,
    Ionicons,
} from '@expo/vector-icons'

export type IconType = "MaterialCommunityIcons" | "MaterialIcons" | "Feather" | 'AntDesign' | "Foundation" | "Ionicons";
export type IconName =
    | React.ComponentProps<typeof MaterialIcons>["name"]
    | React.ComponentProps<typeof MaterialCommunityIcons>["name"]
    | React.ComponentProps<typeof Feather>["name"]
    | React.ComponentProps<typeof AntDesign>["name"]
    | React.ComponentProps<typeof Foundation>["name"]
    | React.ComponentProps<typeof Ionicons>["name"]

interface IconComponentProps {
    iconType: IconType;
}


type Props = IconComponentProps & IconProps<any>

export function IconComponent (props: Props): JSX.Element {
    const {
        iconType,
        style,
        ...otherProps
    } = props
    switch (iconType) {
        case 'MaterialCommunityIcons':
            return <MaterialCommunityIcons {...otherProps} style={style} />
        case 'MaterialIcons':
            return <MaterialIcons {...otherProps} style={style} />

        case 'Feather':
            return <Feather {...otherProps} style={style} />

        case 'AntDesign':
            return <AntDesign {...otherProps} style={style} />

        case 'Foundation':
            return <Foundation {...otherProps} style={style} />

        case 'Ionicons':
            return <Ionicons {...otherProps} style={style} />
        default:
            return <></>
    }
}
