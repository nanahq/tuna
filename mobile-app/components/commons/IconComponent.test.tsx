import {render} from '@testing-library/react-native'
import {IconComponent} from './IconComponent'

describe('<IconComponent />', () => {
    it('Should match snapshot', () => {
        const rendered = render(
            <IconComponent iconType='Feather' name='arrow-left' />
        )
        // @ts-ignore
        expect(rendered.toJSON()).toMatchSnapshot()
    })
})
