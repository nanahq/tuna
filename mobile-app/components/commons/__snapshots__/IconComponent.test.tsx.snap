// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconComponent /> Should match snapshot 1`] = `<Text />`;
